### PR TITLE
Adding metric for other validators' proposal timestamp difference

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
As part of https://whimsical.com/node-status-and-health-metrics-QXjDREqeRDvGDu6xg3J8wx, this PR adds:
```
# TYPE rn_bft_proposal_timestamp_difference_seconds gauge
rn_bft_proposal_timestamp_difference_seconds{key="03fff97bd5755eeea420453a14355235d382f6472f8568a18b2f057a1460297556",component_address="validator_loc1sv7h9eggskmgf7eat8jfd5pvsfx6tf728nwres7zae8694cp5xewdk",} -0.001
```

The interpretation of the above is of course:
"When that validator was creating the proposal, its wallclock was `-0.001` seconds away from my current wallclock."
We are aware that this "clock skew" actually includes the network delay too - but that's all we care about, since we accept/reject proposals based on this clock difference.

### CODE REVIEW

File: core/src/main/java/com/radixdlt/consensus/bft/processor/ProposalTimestampVerifier.java [hunk 1]
```diff
@@ -132,6 +139,15 @@ public void processProposal(Proposal proposal) {
     final var prevTimestamp = proposal.getVertex().parentLedgerHeader().proposerTimestamp();
     final var proposalTimestamp = proposal.getVertex().proposerTimestamp();
 
+    metrics
```
Comment (LukasGasior1) [inline core/src/main/java/com/radixdlt/consensus/bft/processor/ProposalTimestampVerifier.java:142]: I wonder if we should filter out `self` here? It should always be near 0 and could skew the metrics if we e.g. average the diffs

**Review by LukasGasior1 — APPROVED** [review]
just a minor comment, LGTM overall

File: core/src/main/java/com/radixdlt/consensus/bft/processor/ProposalTimestampVerifier.java (hunk 1 above)
Comment (LukasGasior1) [reply core/src/main/java/com/radixdlt/consensus/bft/processor/ProposalTimestampVerifier.java:142]: but I guess we could as well just filter it out when we actually avg :+1:

Comment (jakrawcz-rdx) [reply core/src/main/java/com/radixdlt/consensus/bft/processor/ProposalTimestampVerifier.java:142]: I could.

But we can also calculate `avg_diff_without_self = avg_diff * N/(N-1)`.
(where `N` = number of validators; assuming `self` is near 0 indeed)

Another interpretation of the above fact: the inclusion of `self` here only skews the `avg` metric by a factor of ~`1/N`. It does not skew other interesting metrics, like `max/min`.

And IF the `self`, for some reason, becomes significantly-non-`0`? Then I will appreciate it is in the metrics very much (and will probably go and investigate what causes the delay of proposal processing in the Node internally).

**Pull request merged**